File: 2779-maximum-beauty-of-an-array-after-applying-operation/2779-maximum-beauty-of-an-array-after-applying-operation.py
class Solution:
    def maximumBeauty(self, nums: List[int], k: int) -> int:
        nums.sort() # This make the subsequence problem into the substring problem :)

        answer, l = 0, 0
        for r, rit in enumerate(nums):
            lit = lambda:  nums[l]
            while lit() + k < rit - k:
                l += 1
            answer = max(answer, r - l + 1)
        return answer

File: 2779-maximum-beauty-of-an-array-after-applying-operation/2779-maximum-beauty-of-an-array-after-applying-operation.c
#define max(X, Y) ((X) > (Y) ? (X) : (Y))

int compare(const void *lit, const void *rit) {
  int l = *(int *)(lit);
  int r = *(int *)(rit);

  return l - r;
}

int maximumBeauty(int *nums, int numsSize, int k) {
  qsort(nums, numsSize, sizeof(int), compare);
  int answer = 0;
  int *lit = nums;

  for (int *rit = nums; rit != nums + numsSize; ++rit) {
    while (*lit + k < *rit - k) {
      ++lit;
    }
    answer = max(answer, rit - lit + 1);
  }
  return answer;
}